kani-compiler/src/kani_middle/transform/check_uninit/delayed_ub/instrumentation_visitor.rs

```diff
@@ -44,13 +37,23 @@ impl<'a, 'tcx> TargetFinder for InstrumentationVisitor<'a, 'tcx> {
     fn find_next(
         &mut self,
         body: &MutableBody,
-        bb: BasicBlockIdx,
-        skip_first: bool,
+        source: &SourceInstruction,
     ) -> Option<InitRelevantInstruction> {
-        self.skip_next = skip_first;
-        self.current = SourceInstruction::Statement { idx: 0, bb };
         self.target = None;
-        self.visit_basic_block(&body.blocks()[bb]);
+        match *source {
+            SourceInstruction::Statement { idx, bb } => {
+                let BasicBlock { statements, .. } = &body.blocks()[bb];
+                let stmt = &statements[idx];
+                self.visit_statement(stmt, self.__location_hack_remove_before_merging(stmt.span))
```

Comment:
Do we need to use the `Visitor` trait at this point?

Reply:
Added a commit removing `MirVisitor` dependency